cypress/support/commands.js

```diff
@@ -86,3 +89,216 @@ Cypress.Commands.add('stubFetch', ({ fixture }) => {
+  }
+  return chain;
+}
+
+[
+  'enter',
+  'backspace',
+  ['selectAll', 'selectall'],
+  ['up', 'upArrow'],
+  ['down', 'downArrow'],
+  ['left', 'leftArrow'],
+  ['right', 'rightArrow'],
+].forEach(key => {
+  const [ cmd, keyName ] = typeof key === 'object' ? key : [key, key];
+  Cypress.Commands.add(cmd, { prevSubject: true }, (subject, { shift, times = 1 } = {}) => {
+    const fn = chain => chain.type(`${shift ? '{shift}' : ''}{${keyName}}`);
+    return runTimes(cy.wrap(subject), fn, times);
+  });
+});
+
+// Convert `tab` command from plugin to a child command with `times` support
+Cypress.Commands.add('tabkey', { prevSubject: true }, (subject, { shift, times } = {}) => {
+  const fn = chain => chain.tab({ shift });
+  return runTimes(cy, fn, times).wrap(subject);
+});
+
+Cypress.Commands.add('selection', { prevSubject: true }, (subject, fn) => {
+  cy.wrap(subject)
+    .trigger('mousedown')
+    .then(fn)
+    .trigger('mouseup')
+
+  cy.document().trigger('selectionchange');
+  return cy.wrap(subject);
+});
+
+Cypress.Commands.add('print', { prevSubject: 'optional' }, (subject, str) => {
+  cy.log(str);
+  console.log(`cy.log: ${str}`);
+  return cy.wrap(subject);
+});
+
+Cypress.Commands.add('setSelection', { prevSubject: true }, (subject, query, endQuery) => {
+  return cy.wrap(subject)
+    .selection($el => {
+      if (typeof query === 'string') {
+        const anchorNode = getTextNode($el[0], query);
+        const focusNode = endQuery ? getTextNode($el[0], endQuery) : anchorNode;
+        const anchorOffset = anchorNode.wholeText.indexOf(query);
+        const focusOffset = endQuery ?
+          focusNode.wholeText.indexOf(endQuery) + endQuery.length :
+          anchorOffset + query.length;
+        setBaseAndExtent(anchorNode, anchorOffset, focusNode, focusOffset);
+      } else if (typeof query === 'object') {
+        const el = $el[0];
+        const anchorNode = getTextNode(el.querySelector(query.anchorQuery));
+        const anchorOffset = query.anchorOffset || 0;
+        const focusNode = query.focusQuery ? getTextNode(el.querySelector(query.focusQuery)) : anchorNode;
+        const focusOffset = query.focusOffset || 0;
+        setBaseAndExtent(anchorNode, anchorOffset, focusNode, focusOffset);
+      }
+    });
+});
+
+Cypress.Commands.add('setCursor', { prevSubject: true }, (subject, query, atStart) => {
+  return cy.wrap(subject)
+    .selection($el => {
+      const node = getTextNode($el[0], query);
+      const offset = node.wholeText.indexOf(query) + (atStart ? 0 : query.length);
+      const document = node.ownerDocument;
+      document.getSelection().removeAllRanges();
+      document.getSelection().collapse(node, offset);
+    });
+});
+
+Cypress.Commands.add('setCursorBefore', { prevSubject: true }, (subject, query) => {
+  cy.wrap(subject).setCursor(query, true);
+});
+
+Cypress.Commands.add('setCursorAfter', { prevSubject: true }, (subject, query) => {
+  cy.wrap(subject).setCursor(query);
+});
```

Comment:
Functions above provide naive selection support for cypress, works for our immediate purposes